src/ir/typecache.cpp

```diff
@@ -79,7 +86,7 @@ RecordType* TypeCache::getRecord(RecordParams params) {
     RecordType* r = new RecordType(c,params);
     
     //Inout just needs a single type
-    if (r->isInOut()) {
+    if (r->isInOut() || params.size()==0) {
```

Comment:
This seems like a strange edge case to be supporting, is there a case where you want to have an empty record type? Perhaps that should be an assertion or error?

Reply:
It is actually pretty common. A lot of my testbenches have no inputs or outputs and therefore are just an empty record. They just instantiate the DUT.

Reply:
Ah, that makes sense